runtime/stdlib/cadence_v0.42_to_v1_contract_upgrade_validator.go

```diff
@@ -271,6 +283,19 @@ func (validator *CadenceV042ToV1ContractUpdateValidator) checkTypeUpgradability(
 			break
 		}
 		validator.currentRestrictedTypeUpgradeRestrictions = oldType.Types
+
+		// If the old restricted type is for AnyStruct/AnyResource,
+		// require them to drop the "restricted type".
+		// e.g: `T{I} -> {I}`
+		if restrictedNominalType, isNominal := oldType.LegacyRestrictedType.(*ast.NominalType); isNominal {
+			switch restrictedNominalType.Identifier.Identifier {
+			case "AnyStruct", "AnyResource":
+				break typeSwitch
+			}
+		}
```

Comment:
Does the code already handle the case where the restricted type is inferred, i.e. `{SI}` is given and it means `AnyStruct{SI}`?

Reply:
yeah it should, since the validator works directly on the syntax (and no inferring is involved at this point). I'll add a test just to be sure.

Reply:
adding a test for this isn't much useful; both the old-code and the new-code look the same.
e.g: Old code, with inferred-type is
```cadence
access(all) var a: @{I} 
```
And the required update, is also the same (keeping the restrictions only which is also ` @{I} `).